.github/workflows/ci.yaml

```diff
@@ -0,0 +1,79 @@
+name: CI
+on:
+  pull_request:
+  push:
+      branches:
+        - master
+      tags:
+        - "*.*"
+
+env:
+  MYSQL_USER: root
+  MYSQL_PASSWORD: root
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: corretto
+      - uses: actions/cache@v3
+        with:
+          path: |
+            ~/.ivy2
+            ~/.sbt
+          key: ${{ runner.os }}-sbt-${{ hashFiles('**/*.sbt') }}
+      - name: Start mysql
+        run: sudo /etc/init.d/mysql start
+      - name: Set SBT_OPTS
+        run: bash generate_sbt_opts.sh
+      - name: Build
+        run: sbt package
+      - name: Test
+        run: sbt test
+      - uses: actions/upload-artifact@v3
+        with:
+          name: jars
+          path: '*/target/**/*.jar'
+          retention-days: 1
+  publish:
```

Comment:
I'm not sure. I'd actually expect the sbt cache to be more useful for that. I mostly just copied this from existing sbt projects. I think the upload is mostly for the release step, so that it doesn't have to rebuild the files to upload to github.